hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/ethereum/EthTxData.java

```diff
@@ -193,8 +207,13 @@ public long getAmount() {
     }
 
     public BigInteger getMaxGasAsBigInteger() {
+        long multiple = 1L;
+        if (Arrays.equals(rawTx, FOUNDRY_DETERMINISTIC_DEPLOYER_TRANSACTION)) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**thought:** Perhaps we should check for the `FOUNDRY_DETERMINISTIC_DEPLOYER_TRANSACTION` value only for the `LEGACY_ETHEREUM` type of transactions instead of of all types?‏

Reply:
makes sense.